src/LinearRewardsErc4626.sol

```diff
@@ -207,13 +232,52 @@ abstract contract LinearRewardsErc4626 is ERC4626 {
         totalBalance += amount;
     }
 
+    function _computeFeeRaw(uint256 _assets) private view returns (uint256) {
+        return _assets * mintingFee / FEE_PRECISION;
+    }
+
+    function _computeFeeTotal(uint256 _assets) private view returns (uint256) {
+        return (_assets * mintingFee) / (mintingFee + FEE_PRECISION);
+    }
+
+    function _takeFee(uint256 _feeAmount) private {
+        if (_feeAmount > 0) {
+            asset.safeTransferFrom(msg.sender, FEE_RECIPIENT, _feeAmount);
+            emit FeeTaken(FEE_RECIPIENT, _feeAmount);
+        }
+    }
+
+    function previewDeposit(uint256 _assets) public view override returns (uint256) {
+        uint256 feeAmount = _computeFeeTotal(_assets);
+        return super.previewDeposit(_assets - feeAmount);
+    }
+    
     /// @notice The ```deposit``` function allows a user to mint shares by depositing underlying
     /// @param _assets The amount of underlying to deposit
     /// @param _receiver The address to send the shares to
     /// @return _shares The amount of shares minted
     function deposit(uint256 _assets, address _receiver) public override returns (uint256 _shares) {
         syncRewardsAndDistribution();
-        _shares = super.deposit({ assets: _assets, receiver: _receiver });
+
+        uint256 feeAmount = _computeFeeTotal(_assets);
+
+        uint256 assetsNoFee = _assets - feeAmount;
+
+        // Check for rounding error since we round down in previewDeposit.
+        /// @dev calling super.previewDeposit to ignore feeBPS
+        require((_shares = super.previewDeposit(assetsNoFee)) != 0, "ZERO_SHARES");
```

Comment:
# Q - Shouldn't this be `_assets` since you already account for the fee on `previewDeposit`?